connector_elasticsearch/components/adapter.py

```diff
@@ -79,10 +85,16 @@ def delete(self, binding_ids):
                 "_id": binding_id,
             }
             dataforbulk.append(action)
-
-        res = elasticsearch.helpers.bulk(es, dataforbulk)
-        # checks if number of indexed object and object in datas are equal
-        return len(binding_ids) - res[0] == 0
+        try:
+            elasticsearch.helpers.bulk(es, dataforbulk)
+        except elasticsearch.helpers.errors.BulkIndexError as e:
+            # if the document we are trying to delete does not exist,
+            # we can consider deletion a success (there is nothing to do).
+            if not _is_delete_nonexistent_documents(e):
+                raise e
+            msg = "Trying to delete non-existent documents. Ignored: %s"
+            _logger.info(msg, e)
+        return True
```

Comment:
nit nit: if no one cares about the result, why the return ?

Which make me think, as we often return True to please xmlrpc, can these Component methods be called via xmlrpc ? In which case exposing this `delete` might be dangerous.

Reply:
Because the tests check for it, so I would not change the function interface without a good reason.
For your second question, I doubt it is the case as they obviously exist to implement private logic, but you should ask someone competent.

Reply:
Yes, the [component interface](https://github.com/OCA/search-engine/blob/0d166ecbb6a2f00647085eadc12ad81f86a1b06b/connector_search_engine/components/adapter.py#L21) does not talk about returning a boolean nor does the [algolia implementation](https://github.com/OCA/search-engine/blob/0d166ecbb6a2f00647085eadc12ad81f86a1b06b/connector_algolia/components/adapter.py#L59), so better return None and change the test.